GVRf/Framework/framework/src/main/java/org/gearvrf/GVRBitmapImage.java

```diff
@@ -125,9 +129,27 @@ public GVRBitmapImage(GVRContext gvrContext, int width, int height, byte[] grays
      */
     public void setBitmap(Bitmap bmap)
     {
-        NativeBitmapImage.updateFromBitmap(getNative(), bmap, bmap.hasAlpha());
+        Bitmap.Config config = bmap.getConfig();
+
+        //if a bitmap of format other than supported format is found,
+        //try to create a new bitmap from it of config ARGB_8888.
+        if (!supportedConfigs.contains(config))
+            bmap = getBitmapSupported(bmap);
+
+        NativeBitmapImage.updateFromBitmap(getNative(), bmap, bmap.hasAlpha(), bmap.getConfig().name());
+    }
+
+    private Bitmap getBitmapSupported(Bitmap orig)
+    {
+        Bitmap supBitmap = Bitmap.createBitmap( orig.getWidth(), orig.getHeight(), Bitmap.Config.ARGB_8888 );
+        Canvas c = new Canvas(supBitmap);
+        Paint p = new Paint();
+        c.drawBitmap(orig,0,0,p);
+        orig.recycle();
```

Comment:
Please don't recycle the original immediately. It won't be recycled if conversion is not necessary so lets not introduce special case side-effects.